eCAL not buildable with Protobuf 3.20

**Describe the bug**
Currently eCAL no longer builds against Protobuf 3.20 since `google::protobuf::string` was removed.
This currently causes the eCAL MacOS build to fail.

```
/Users/runner/work/ecal/ecal/samples/cpp/misc/proto_dyn/src/proto_dyn.cpp:79:73: error: no type named 'string' in namespace 'google::protobuf'; did you mean 'std::string'?
void ProcProtoType(const std::string& group_, const std::string& name_, google::protobuf::string value_, size_t index_)
                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~
                                                                        std::string
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/iosfwd:240:65: note: 'std::string' declared here
typedef basic_string<char, char_traits<char>, allocator<char> > string;
                                                                ^
```

**Additional context**
The following files are affected:
https://github.com/continental/ecal/blob/a4de58bf65b8e5a3ef057e9c63513695ca143403/samples/cpp/misc/proto_dyn/src/proto_dyn.cpp#L79
https://github.com/continental/ecal/blob/a4de58bf65b8e5a3ef057e9c63513695ca143403/testing/contrib/ecalproto/dynproto_test/src/dynproto_test.cpp#L150

We can remove this function, and directly call `ProcString` instead.

--- a/samples/cpp/misc/proto_dyn/src/proto_dyn.cpp
+++ b/samples/cpp/misc/proto_dyn/src/proto_dyn.cpp
@@ -76,7 +76,7 @@ void ProcProtoType(const std::string& group_, const std::string& name_, bool val
   ProcValue(group_, name_, double(value_), index_);
 }
 
-void ProcProtoType(const std::string& group_, const std::string& name_, google::protobuf::string value_, size_t index_)
+void ProcProtoType(const std::string& group_, const std::string& name_, const std::string& value_, size_t index_)
 {
   ProcString(group_, name_, value_, index_);
 }

--- a/testing/contrib/ecalproto/dynproto_test/src/dynproto_test.cpp
+++ b/testing/contrib/ecalproto/dynproto_test/src/dynproto_test.cpp
@@ -147,7 +147,7 @@ void ProcProtoType(const std::string& group_, const std::string& name_, bool val
   ProcValue(group_, name_, double(value_), index_);
 }
 
-void ProcProtoType(const std::string& group_, const std::string& name_, google::protobuf::string value_, size_t index_)
+void ProcProtoType(const std::string& group_, const std::string& name_, const std::string& value_, size_t index_)
 {
   ProcString(group_, name_, value_, index_);
 }

